docs/modules/guides/partials/update-macro-api.adoc

```diff
@@ -0,0 +1,75 @@
+== Update Macro Processors
+
+In the earlier version of AsciidoctorJ (2.5.x), the signature of Macro Processors required the implementation to return an unrestricted Object.
+However, the specific type that a Macro Processor had to return was unclear, making it difficult to understand the necessary implementation steps.
+
+To address this issue, AsciidoctorJ 3.x.x has introduced the following changes:
+
+`InlineMacroProcessor::process`::
+The implementation of this method now needs to return a PhraseNode.
+Please update your code accordingly to ensure that the returned value is of type PhraseNode.
+`BlockMacroProcessor::process`::
+The implementation of this method now needs to return a StructuralNode.
+Make sure to modify your implementation to return a StructuralNode as required.
+
+Moreover, there was a problem with the first parameter of the process method in InlineMacroProcessors.
+Previously, it incorrectly expected a PhraseNode as the first parameter.
```

Comment:
I like to make it clear when I am pointing to a class or method using backticks (`).

```suggestion
Previously, it incorrectly expected a `PhraseNode` as the first parameter.
```